### PR TITLE
[master < T0766-MG] Fix ORDER BY and temporal types

### PR DESCRIPTION
**Pull request by antonio2368**
[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Update [changelog](https://docs.memgraph.com/memgraph/changelog)

### CODE REVIEW

**Review by kostasrim — APPROVED** [review]
:rocket:

File: src/query/common.cpp [hunk 1]
```diff
@@ -43,6 +43,14 @@ bool TypedValueCompare(const TypedValue &a, const TypedValue &b) {
...
     case TypedValue::Type::String:
       return a.ValueString() < b.ValueString();
+    case TypedValue::Type::Date:
+      return a.ValueDate() < b.ValueDate();
+    case TypedValue::Type::LocalTime:
+      return a.ValueLocalTime() < b.ValueLocalTime();
+    case TypedValue::Type::LocalDateTime:
+      return a.ValueLocalDateTime() < b.ValueLocalDateTime();
+    case TypedValue::Type::Duration:
+      return a.ValueDuration() < b.ValueDuration();
```
Comment (antaljanosbenjamin) [inline src/query/common.cpp:53]: Why not to delete the default branch?

Comment (antonio2368) [reply src/query/common.cpp:53]: I forgot, I should delete it.

**Review by jbajic — APPROVED** [review]

**Comment by antaljanosbenjamin** [comment]
Maybe one more question: do we want to allow the comparison between date and datetime? witthout this, it feels like a bit incomplete (a date can be considered as a datetime with the time part zerod out).

**Comment by antonio2368** [comment]
> Maybe one more question: do we want to allow the comparison between date and datetime? witthout this, it feels like a bit incomplete (a date can be considered as a datetime with the time part zerod out).

If we do that we should allow the same thing in the comparison operator for TypedValues which we don't for now.
We can definitely discuss this.

**Pull request merged**